Improved documentation

h4yn0nnym0u5e opened the pull request:
Based on various threads on the PJRC forum, an attempt to make it a bit clearer how to get things working. Always supposing anyone bothers to read it, of course...

Now also fixes a significant bug, and gives independent control over greeting and message levels.

h4yn0nnym0u5e committed:
Add images folder with helpful stuff

h4yn0nnym0u5e committed:
Improved ReadMe

Shows expected file structure, IDE appearance and where to get the MTP library

h4yn0nnym0u5e committed:
Corrected library name

h4yn0nnym0u5e committed:
Added image for MTP_Teensy library location

h4yn0nnym0u5e committed:
Show MTP_Teensy library location in ReadMe

eanplatter commented on README.md:
```diff
@@ -1,16 +1,22 @@
-Unfortunately, "installation" is a bit messy, but you only have to do it once (unless you update your Arduino IDE or Teensyduino). The guestbook sketch folder contains two Arduino architecture configuration files, `boards.local.txt` and`platform.txt`. These must be used to update your existing configuration files; in Windows systems these will typically be in `C:\Program Files (x86)\Arduino\hardw ...
+Unfortunately, "installation" is a bit messy, but you only have to do it once (unless you update your Arduino IDE or Teensyduino).
+#### Tools menu option
+ The guestbook sketch folder contains two Arduino architecture configuration files, `boards.local.txt` and`platform.txt`. These must be used to update your existing configuration files; in Windows systems these will typically be in `C:\Program Files (x86)\Arduino\hardware\teensy\avr` [todo: where are they on Linux and Mac?].
```
Hey @h4yn0nnym0u5e  I was going to make a small update to the README to include the config path on Macs and saw your PR so I thought I could just suggest a change instead of make my own PR.

```suggestion
 The guestbook sketch folder contains two Arduino architecture configuration files, `boards.local.txt` and`platform.txt`. These must be used to update your existing configuration files; in Windows systems these will typically be in `C:\Program Files (x86)\Arduino\hardware\teensy\avr` or `/Applications/Teensyduino.app/Contents/Java/hardware/teensy/avr` on Mac [todo: where are they on Linux?].
```

DaT729 replied:
How do I fix this error code on a mac? I am trying my best to understand it.


" 
Arduino: 1.8.19 (Mac OS X), TD: 1.57, Board: "Teensy 4.0, MTP Disk (Experimental), 600 MHz, Faster, US English"











Multiple libraries were found for "SD.h"
 Used: /private/var/folders/_2/tl13lpw51jj5_y71zm7qd__40000gp/T/AppTranslocation/3E053CC5-9134-4F53-AC58-809E7713ABB8/d/Teensyduino.app/Contents/Java/hardware/teensy/avr/libraries/SD
 Not used: /private/var/folders/_2/tl13lpw51jj5_y71zm7qd__40000gp/T/AppTranslocation/3E053CC5-9134-4F53-AC58-809E7713ABB8/d/Teensyduino.app/Contents/Java/libraries/SD
/Users/odastaff/Desktop/Audiobook/Audiobook.ino:30:24: fatal error: MTP_Teensy.h: No such file or directory
compilation terminated.
Error compiling for board Teensy 4.0.


This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

markmak77 replied:
@DaT729 , that's not an issue relating to mac, but to a missing library (people on Windows also get this error). The issue and its solution can be found on the "issues" page of this project: https://github.com/playfultechnology/audio-guestbook/issues/63

h4yn0nnym0u5e committed:
Minor correction in README.md

h4yn0nnym0u5e committed:
Fix issue with hanging up early causing huge recordings

See e.g. https://forum.pjrc.com/threads/70553-Teensy-4-0-based-Audio-Guestbook?p=325434&viewfull=1#post325434

The problem is that the wait() function throws away the ability to detect hook switch edges. Prior to this change, hanging up in the "Wait a second for users to put the handset to their ear" period resulted in recording continuing until the next hang-up, potentially filling the SD card.

The change simply uses the hook switch level to determine whether to abandon the recording.

h4yn0nnym0u5e commented:
Added commit e0307f3 which fixes a bug that's caused a fair few problems, e.g. https://forum.pjrc.com/threads/70553-Teensy-4-0-based-Audio-Guestbook?p=325434&viewfull=1#post325434

Expect it'll get ignored, though.

h4yn0nnym0u5e committed:
Allow independent setting of greeting and message playback levels

Where the greeting is not recorded using the phone hardware, it may have a very different level. Use a separate playback object and mixer channel for the "greeting.wav" playback, so the level can be adjusted if needs be.

h4yn0nnym0u5e committed:
Add AVC and timeout

Enabled the SGTL5000's AVC (auto volume control) audio processing to the mic input, to try to help with the possible wide range of speaking (shouting!) voices the system may encounter.

Added a 2-minute timeout so the SD card won't fill up if left off-hook. Defined by MAX_RECORDING_TIME_MS, change to suit your needs.

h4yn0nnym0u5e committed:
Add ability to load configuration from file

This allows the user to have a pre-built binary, depending only on the Teensy model in use, and configuring switch polarity and volume levels using the gbkcfg.txt file.

h4yn0nnym0u5e committed:
Tidied up

h4yn0nnym0u5e committed:
Fix issue with defaulting to non-standard SGTL I²C address

Also make it configurable, along with max recording length
